src/Stratis.Bitcoin/Consensus/ConsensusManager.cs

```diff
@@ -513,8 +513,23 @@ private async Task<ConnectBlocksResult> FullyValidateLockedAsync(ChainedHeader n
             // If the new block is not on the current chain as our current consensus tip then rewind consensus tip to the common fork.
             bool isExtension = fork == oldTip;
 
+            List<ChainedHeaderBlock> disconnectedBlocks = null;
+
             if (!isExtension)
             {
+                // Save blocks that will be disconnected in case we will need to reconnect them (this might happen if connection of a new chain fails).
```

Comment:
This whole new code can go into separated method

Reply:
Maybe even better, you could add a new argument to RewindToForkPointAsync, and if that argument would be true, that method would return the list of disconnected blocks because it essentially has those blocks because it signals them. Or maybe it could return the list every time because it is not that common that we don't need the result